python/ql/lib/semmle/python/regex.qll

```diff
@@ -204,8 +204,7 @@ abstract class RegexString extends Expr {
     |
       end = inner_end + 1 and
       inner_end > inner_start and
-      this.nonEscapedCharAt(inner_end) = "]" and
-      not exists(int mid | this.nonEscapedCharAt(mid) = "]" | mid > inner_start and mid < inner_end)
+      inner_end = min(int i | this.nonEscapedCharAt(i) = "]" and inner_start < i)
```

Comment:
We can now also delete the line `inner_end > inner_start` since it is implied by the definition of `inner_end`.

Reply:
Done! Good catch!